### PR TITLE
chore: integrate che-docs into overall release process

### PR DESCRIPTION
**Pull request by mkuznyetsov**
## What does this pull request change?
Add a workflow for performing automated tasks related to Eclipse Che release.
This job can be triggered manually, as well as it will be called automatically at the end of the general automated release workflow.
- create 7.x.y branch and tag
- create pull-request to main and bugfix branch updating antora files.
- create pull-request to publication-builder branch.
Screenshot from 2024-03-19 11-32-45

additionally, create and tag release image for better visibility of history on quay.io (to determine which image is related to which version of che)

## What issues does this pull request fix or reference?
https://github.com/eclipse/che/issues/22764

## Pull Request checklist

The author and the reviewers validate the content of this pull request with the following checklist, in addition to the [automated tests](code_review_checklist.adoc).

- Any procedure:
  - [ ] Successfully tested.
- Any page or link rename:
  - [ ] The page contains a redirection for the previous URL.
  - Propagate the URL change in:
    - [ ] Dashboard [default branding data](https://github.com/eclipse-che/che-dashboard/blob/main/packages/dashboard-frontend/src/services/bootstrap/branding.constant.ts)
- [ ] Builds on [Eclipse Che hosted by Red Hat](https://workspaces.openshift.com).
- [ ] the *`Validate language on files added or modified`* step reports no vale warnings.

### CODE REVIEW

**Comment by mkuznyetsov** [comment]
@deerskindoll is there a need to tag an push container image for che-docs? is it something that needs to be used?

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      - name: Set up yq
+        run: |
+          python -m pip install --upgrade pip
+          pip install yq
+      - name: Create Release
+        run: |
+          git config --global user.name "Mykhailo Kuznietsov"
+          git config --global user.email "mkuznets@redhat.com"
+          export GITHUB_TOKEN=${{ secrets.CHE_BOT_GITHUB_TOKEN }}
+          /bin/bash make-release.sh --version ${{ github.event.inputs.version }} --trigger-release
```
Comment (ibuziuk) [inline .github/workflows/release.yml]: no new line

**Comment by deerskindoll** [comment]
> @deerskindoll is there a need to tag an push container image for che-docs? is it something that needs to be used?

the only thing we do for releases is described here: https://github.com/eclipse-che/che-docs/blob/main/RELEASE.adoc: 
- create a new branch for the release
- update antora.yaml in `main` and the release branch
- update publication-builder-antora-playbook.yml in the `publication-builder` branch

File: make-release.sh [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+if [[ $TAG_RELEASE -eq 1 ]]; then
+  echo "[INFO] Creating release tag"
+  git checkout "${BRANCH}" && git pull origin "${BRANCH}" || true
+  git tag "${VERSION}"
+  git push origin "${VERSION}" || true
+fi
+
+if [[ ${USE_TMP_DIR} -eq 1 ]]; then
+  rm -fr /tmp/$tmpdir
+fi
```
Comment (ibuziuk) [inline make-release.sh]: no new line

**Review by deerskindoll — APPROVED** [review]
lgtm

**Pull request merged**